Makefile

```diff
@@ -36,18 +37,20 @@ all: $(BIN) data
 clean:
 	rm -rf $(OBJECTS_DIR)
 	rm -f $(BIN)
-	rm -f $(TEST_BIN)
+	rm -f $(TEST_BIN) $(TEST_OBJECTS)
 	rm -f $(JSON_DATA)
 	rm -rf data
 
-data: $(SERVER_JAR_EXTRACTED)
-
 run: all
 	./$(BIN)
 
-$(SERVER_JAR_EXTRACTED):
+data: $(SERVER_JAR_EXTRACTED)
+
+data/server.jar:
```

Comment:
This was intentionally NOT split, as it will ensure the .jar is re-downloaded whenever the version number changes.

I agree that splitting makes sense, but then we should add the version number into the downloaded filename (and extract it to a variable to avoid duplication).